openmeter/entitlement/driver/parser.go

```diff
@@ -54,7 +54,13 @@ func (parser) ToMetered(e *entitlement.Entitlement) (*api.EntitlementMetered, er
 		CurrentUsagePeriod:     *mapPeriod(e.CurrentUsagePeriod),
 		LastReset:              metered.LastReset,
 		PreserveOverageAtReset: convert.ToPointer(metered.PreserveOverageAtReset),
-	}, nil
+	}
+
+	if e.SubscriptionManaged {
```

Comment:
Instead of this, cloud you validate that the metered.Metadata is not nil, clone the metered.Metadata before assembling the type and set these values on the metadata?